prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -385,5 +388,69 @@ public async Task<ActionResult> sendRemoteUsersEmail(int siteId)
             return NoContent();
         }
 
+        // POST: api/Sites/5/approval
+        /// <summary>
+        /// Approved a site, setting it's approval date
+        /// </summary>
+        /// <param name="siteId"></param>
+        [HttpPost("{siteId}/approval", Name = nameof(ApproveSite))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<Site>> ApproveSite(int siteId)
+        {
+            if (!User.IsAdmin())
```

Comment:
rather than checking this in the method, you can add the 
`[Authorize(Policy = AuthConstants.ADMIN_POLICY)]`
header to this method.